plugins/sidebar/overlays/download.js

```diff
@@ -8,102 +8,32 @@ const Button = require('react-material/components/Button');
 const Loader = require('react-loader');
 const Progress = require('./progress');
 
-const styles = require('../styles');
-
-class DownloadOverlay extends React.Component {
-  constructor(){
-    this.state = {
-      devicePath: null,
-      searching: false,
-      selectedDevice: null,
-      devices: [],
-      progress: 0
-    };
-
-    this.onAccept = this.onAccept.bind(this);
-    this.onCancel = this.onCancel.bind(this);
-    this.reloadDevices = this.reloadDevices.bind(this);
-    this.updateSelected = this.updateSelected.bind(this);
-    this.updateProgress = this.updateProgress.bind(this);
-  }
-
-  componentDidMount(){
-    this.reloadDevices();
-  }
-
-  onAccept(evt){
-    this.download(this.state.selectedDevice);
-  }
-
-  onCancel(evt){
-    if(typeof this.props.onCancel === 'function'){
-      this.props.onCancel(evt);
-    }
-  }
-
-  download(device){
-
-    const { irken, handleError, handleSuccess } = this.props;
-
-    const { toast, workspace, logger, overlay } = irken;
-
-    const name = workspace.filename.deref();
-    const source = workspace.current.deref();
-
-    if(!device){
-      return;
-    }
+const DeviceActions = require('../../../src/actions/DeviceActions.js');
+const DeviceStore = require('../../../src/stores/DeviceStore.js');
 
-    const board = irken.getBoard(device);
+const { createContainer } = require('sovereign');
 
-    board.removeListener('terminal', logger);
-
-    board.on('progress', this.updateProgress);
+const styles = require('../styles');
 
-    board.compile(source)
-      .tap(() => logger.clear())
-      .then((memory) => board.bootload(memory))
-      .then(() => board.on('terminal', logger))
-      .tap(() => toast.clear())
-      .tap(() => handleSuccess(`'${name}' downloaded successfully`))
-      .catch(handleError)
-      .finally(() => {
-        overlay.hide();
-        board.removeListener('progress', this.updateProgress);
-        this.setState({ progress: 0 });
-      });
+class DownloadOverlay extends React.Component {
 
-  }
+  constructor(){
 
-  updateProgress(progress){
-    this.setState({ progress: progress});
-  }
+    this._onAccept = this._onAccept.bind(this);
+    this._onCancel = this._onCancel.bind(this);
+    this._onReloadDevices = this._onReloadDevices.bind(this);
+    this._onUpdateSelected = this._onUpdateSelected.bind(this);
 
-  updateSelected(device){
-    this.setState({
-      devicePath: device.path,
-      selectedDevice: device
-    });
   }
 
-  reloadDevices(){
-    const irken = this.props.irken;
-    const scanOpts = {
-      reject: [
-        /Bluetooth-Incoming-Port/,
-        /Bluetooth-Modem/,
-        /dev\/cu\./
-      ]
-    };
-    this.setState({ devicePath: null, searching: true });
-    irken.scanBoards(scanOpts)
-      .then((devices) => this.setState({ devices: devices, searching: false }));
+  componentDidMount(){
+    this._onReloadDevices();
```

Comment:
i think fetching initial data should be moved outside of the `componentDidMount` and into the store's constructor